sklearn/model_selection/tests/test_search.py

```diff
@@ -1208,15 +1202,68 @@ def compare_refit_methods_when_refit_with_acc(search_multi, search_acc, refit):
 
 @pytest.mark.filterwarnings("ignore:The parameter 'iid' is deprecated")  # 0.24
 @pytest.mark.parametrize('search_cv', [RandomizedSearchCV, GridSearchCV])
-def test_search_cv_score_samples_method(search_cv):
-    X, y = make_classification(n_samples=50, n_features=4, random_state=42)
-    clf = search_cv(MockClassifier(), {'foo_param': [0, 2]},
-                    scoring='precision', refit=True)
+def test_search_cv_score_samples_error(search_cv):
+    if search_cv == RandomizedSearchCV:
+        clf = search_cv(estimator=DecisionTreeClassifier(),
+                        param_distributions={'max_depth': [5, 10]})
+    else:
+        clf = search_cv(estimator=DecisionTreeClassifier(),
+                        param_grid={'max_depth': [5, 10]})
+
+    X, y = make_blobs(n_samples=100, n_features=4, random_state=42)
     clf.fit(X, y)
-    # Test that the score_samples method on *SearchCV yields same results as
-    # applying score_samples from the estimator itself directly (best param is
-    # foo_param=1)
-    assert_allclose(clf.score_samples(X), MockClassifier(foo_param=2).score_samples(X))
+
+    # Make sure to error out when underlying estimator does not implement
+    # the method `score_samples`
+    err_msg = "AttributeError: 'DecisionTreeClassifier' object has " \
+              "no attribute 'score_samples'"
+    assert_raise_message(AttributeError, err_msg, clf.score_samples, X)
```

Comment:
It seems that the error here is not caught and the test fails. How can I catch the error?
`AttributeError: 'DecisionTreeClassifier' object has no attribute 'score_samples'"`

Reply:
Try:

```py
    err_msg = ("'DecisionTreeClassifier' object has no attribute "
               "'score_samples'")
    with pytest.raises(AttributeError, match=err_msg):
        clf.score_samples(X)
```